code/modules/mob/living/silicon/robot/component.dm

```diff
@@ -21,9 +21,19 @@
 
 /datum/robot_component/proc/install()
 /datum/robot_component/proc/uninstall()
+	//Integrated Lights removed? Lights out.
+	if(istype(src,/datum/robot_component/integrated_light))
+		var/mob/living/silicon/robot/R = src.owner
+		toggled = 0
+		R.set_light(0)
 
 /datum/robot_component/proc/destroy()
 	if(wrapped)
+		//Integrated Lights removed? Lights out.
+		if(istype(src,/datum/robot_component/integrated_light))
+			var/mob/living/silicon/robot/R = src.owner
+			toggled = 0
+			R.set_light(0)
```

Comment:
Again this isn't okay at all, this should not be anywhere near these central procs. Override it on the integrated light if you absolutely have to.